### PR TITLE
feat: Adds B2B Customer ID

### PR DESCRIPTION
**Pull request by ataideverton**
## What's the purpose of this pull request?

As we develop more b2b features inside the `faststore`, we need to be able to identify when a customer is a `b2b` customer and if he is member of an Organization (Normally refered to as a Buyer Organization). This PR addresses the first step to do so.

For that, we added a `b2b` section to the `session` hook and mutation and inside of it, a `customerId` field, that informs us of what BuyerOrganization that customer is a part of.

Why we decided to do this inside FastStore and not as an override or extra request: The information comes from the same `sessions` api `Faststore` already uses, we just need to bring an additional field. This will not impact B2C stores, for them, it will only be an empty objetct 

## How it works?

On a B2B account that fulfills several conditions (Some specific apps, some specific versions of apps, a feature flag), the `sessions` api will return the `customerId` field on the `authentication` namespace and that will be inserted on the internal session object, like so: 

Account:  `b2bfaststoredev`: 
image

On a B2C account, or any account that doesn't fit all the criteria, the field will be empty and should not interfere with anything.

Account: `storeframework`
image

## How to test it?

To make validation easier, we have 2 previews (You will need to be logged in to see it working, as they are previews, you will need to edit the cookie): 
[B2B Preview - Session will print on console every few seconds, only on the /self-management page](https://sfj-45b4ae2--b2bfaststoredev.preview.vtex.app/self-management)

[B2C Preview - Has a button on navbar to print the session on console](https://sfj-342a275--starter.preview.vtex.app/)
The button and the `faststore/cli` dependency are the only changes here

The PRs for these previews: 
[B2B](https://github.com/vtex-sites/b2bfaststoredev.store/pull/91)
[B2C](https://github.com/vtex-sites/starter.store/pull/509)

### CODE REVIEW

**Review by lariciamota — COMMENTED** [review]
Nice! Just left a few small comments

ps.: The build of one of the preview PRs failed, could you check it? Also, could you update the starter.store with the main branch and use the cli package instead of core? So I can test it 😄 Thanks!

File: packages/core/faststore.config.default.js [hunk 1]
```diff
@@ -36,6 +36,7 @@ module.exports = {
     addressType: null,
     postalCode: null,
     geoCoordinates: null,
+    b2b: null,
```
[the remark this answers is not in the document]
Comment (ataideverton) [reply packages/core/faststore.config.default.js:39]: Yeah, the type check breaks without it.

On `packages/core/src/sdk/session/index.ts` we have a StoreConfig that is imported from the merge of this guy and the customer config file. 

Without this, the types don't match :/

image

File: packages/sdk/src/session/index.ts [hunk 2]
```diff
@@ -28,6 +28,10 @@ export interface Person {
   familyName: string
 }
 
+export interface b2bInfo {
```
[the remark this answers is not in the document]
Comment (ataideverton) [reply packages/sdk/src/session/index.ts]: Done!

File: packages/core/src/sdk/session/index.ts [hunk 3]
```diff
@@ -42,6 +42,9 @@ export const mutation = gql(`
         givenName
         familyName
       }
+      b2b{
```
[the remark this answers is not in the document]
Comment (ataideverton) [reply packages/core/src/sdk/session/index.ts]: Done!

**Comment by ataideverton** [comment]
> Nice! Just left a few small comments
> 
> ps.: The build of one of the preview PRs failed, could you check it? Also, could you update the starter.store with the main branch and use the cli package instead of core? So I can test it 😄 Thanks!

The previews were kind of out outdated and the build broke because a few things changed, updated both of them and all should be working as intended right now

**Review by lariciamota — APPROVED** [review]

**Review by hellofanny — APPROVED** [review]

**Pull request merged**